### PR TITLE
Integration tests as a CI step - Closes #704

### LINKED ISSUE #704: Integration tests as a CI step
Include integration test as a new Jenkins stage

### PR DESCRIPTION
**Pull request by MaciejBaj**
- create separate Jenkins job using node-6
- go back to `forging.force = false` enable forging by sending HTTP requests

Closes #704

### CODE REVIEW

File: Jenkinsfile.integration [hunk 1]
```diff
@@ -0,0 +1,71 @@
+node('node-06'){
+  lock(resource: "core-integration-tests", inversePrecedence: true) {
```
Comment (4miners) [inline Jenkinsfile.integration]: tabs instead of spaces

File: Jenkinsfile.integration [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        error('Stopping build, cleanup failed')
+      }
+    }
+
+    stage ('Set milestone') {
+      milestone 1
+      currentBuild.result = 'SUCCESS'
+    }
+  }
+}
```
Comment (4miners) [inline Jenkinsfile.integration]: newline

File: test/integration/peers.integration.js [hunk 3]
```diff
@@ -204,8 +202,76 @@ function recreateDatabases (done) {
...
+				connectedTo += 1;
+				if (connectedTo === testNodeConfigs.length) {
+					done();
+				}
+			});
+			socket.on('error', function (err) {});
+			socket.on('connectAbort', function (err) {
+				done('Unable to establish WS connection with ' + testNodeConfig.ip + ':' + testNodeConfig.port);
+			});
+		}, 1000);
```
Comment (4miners) [inline test/integration/peers.integration.js]: that `1000` should be one line below (expected: `setTimeout(function () {...}, 1000);` now `testNodeConfigs.forEach(function () {...}, 1000);`

**Comment by Isabello** [comment]
I have pushed an update to the jenkins file to include the required hooks to flag this build process as its own set of tests that will be required to pass.

**Comment by Isabello** [comment]
There is still an issue in this PR with jenkins improperly submitting pr-merge, i will continue on this.

File: test/integration/peers.integration.js [hunk 4]
```diff
@@ -204,8 +202,78 @@ function recreateDatabases (done) {
...
+				}
+			});
+			enableForgingPromises.push(enableForgingPromise);
+		});
+	});
+
+	Promise.all(enableForgingPromises).then(function () {
+		done();
+	}).catch(function () {
+		done('Error during registering delegates using http requests');
```
Comment (karmacoma) [inline test/integration/peers.integration.js]: `Failed to enable forging on delegates`

File: test/integration/peers.integration.js [hunk 5]
```diff
@@ -204,8 +202,78 @@ function recreateDatabases (done) {
 	});
 }
 
+function registerDelegatesUsingHttpRequests (done) {
```
Comment (karmacoma) [inline test/integration/peers.integration.js]: `enableForgingOnDelegates`

File: test/integration/peers.integration.js [hunk 6]
```diff
@@ -204,8 +202,78 @@ function recreateDatabases (done) {
...
+			socket.on('connect', function () {
+				sockets.push(socket);
+				connectedTo += 1;
+				if (connectedTo === testNodeConfigs.length) {
+					done();
+				}
+			});
+			socket.on('error', function (err) {});
+			socket.on('connectAbort', function (err) {
+				done('Unable to establish WS connection with ' + testNodeConfig.ip + ':' + testNodeConfig.port);
```
Comment (karmacoma) [inline test/integration/peers.integration.js]: `Failed to establish WS connection with`

File: test/integration/peers.integration.js (hunk 3 above)
Comment (MaciejBaj) [inline test/integration/peers.integration.js]: ofc, thank you

File: test/integration/fixtures/syncModes.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+/**
```
Comment (diego-G) [inline test/integration/fixtures/syncModes.js]: Why an empty file with that comment?

**Comment by diego-G** [comment]
I don't this bullet from the description addressed `go back to forging.force = false enable forging by sending HTTP requests` why @MaciejBaj

File: test/integration/fixtures/syncModes.js (hunk 7 above)
Comment (MaciejBaj) [reply test/integration/fixtures/syncModes.js]: oh, my mistake, thanks

File: test/integration/peers.integration.js [hunk 8]
```diff
@@ -223,76 +277,40 @@ describe('integration', function () {
 	});
 
 	before(function (done) {
```
Comment (diego-G) [inline test/integration/peers.integration.js]: I would like to merge before blocks into one.

**Pull request merged**